src/main/java/seedu/duke/logic/BudgetLogic.java

```diff
@@ -29,22 +32,22 @@ public BudgetLogic(Budget budget, AppUi ui) {
      * Sets the budget if it has not been set. If the budget is already set,
      * prompts the user to confirm whether they want to modify it.
      */
-    public void setBudget() {
+    public void setBudget(FinancialList financialList) throws FinanceBuddyException {
         if (!budget.isBudgetSet()) {
             ui.displaySetBudgetMessage();
-            handleSetBudget();
+            handleSetBudget(financialList);
         } else {
             System.out.println("Your current budget is: " + budget.getBudgetAmount());
```

Comment:
Maybe this can also be extracted as a variable/constant?

Reply:
I think for this I can extract it out to the ui class to handle the displaying of messages?